server/controllers/Comments.js

```diff
@@ -107,6 +107,69 @@ class Comments {
       serverError(res);
     }
   }
+
+  /**
+   * @name delete
+   * @async
+   * @static
+   * @memberof Comments
+   * @param {Object} req express request object
+   * @param {Object} res express response object
+   * @returns {JSON} JSON object with message on deleted message
+   */
+  static async delete(req, res) {
+    try {
+      const { slug, id } = req.params;
+      const { id: userId } = req.user;
+      const article = await Article.findBySlug(slug);
+      if (!article) {
+        return serverResponse(res, 404, { error: 'article not found' });
+      }
+      const articleId = article.id;
+      const deleteComment = await Comment.destroy({
+        where: { id, articleId, userId }
+      });
+      if (!deleteComment) {
```

Comment:
@ayodejiAA Smart implementation! However, I feel your code will be more intuitive and readable if you separate the logic of deleting a comment from the logic of checking if an article exists or not.

Reply:
Thanks,  @abdulfatai360 Basically, the logic is to avoid additional query to the database. if there is no comment to delete, the response will be 0.